internal/kafka/utils.go

```diff
@@ -175,13 +172,9 @@ func topicNameStrategy(topic, mode string) string {
 	return fmt.Sprintf("%s-%s", topic, mode)
 }
 
-func getCmkClusterNetwork(cluster *cmkv2.CmkV2Cluster, ctx *config.Context, client *ccloudv2.Client) string {
-	if isDedicated(cluster) && cluster.Spec.Network.GetId() != "" {
-		network, _ := client.GetNetwork(ctx.GetCurrentEnvironment(), cluster.Spec.Network.GetId())
-
-		if public := slices.Contains(network.Spec.GetConnectionTypes().Items, "PUBLIC"); !public {
-			return cluster.Spec.Network.GetId()
-		}
+func getCmkClusterNetwork(cluster *cmkv2.CmkV2Cluster) string {
+	if isDedicated(cluster) && cluster.Spec.Network != nil {
```

Comment:
If `Network` is `nil`, `GetId()` will return `""`.
```suggestion
	if isDedicated(cluster) {
```

Reply:
Do we even need to make this check? I thought you said only dedicated clusters have a network